diesel/src/query_dsl/save_changes_dsl.rs

```diff
@@ -57,10 +57,64 @@ where
 }
 
 pub trait SaveChangesDsl<Conn> {
-    /// Sugar for types which implement both `AsChangeset` and `Identifiable`
-    ///
-    /// `foo.save_changes(&conn)` is equivalent to
-    /// `update(foo::table().find(foo.id())).set(&foo).get_result(&conn)`
+/// Sugar for types which implement both `AsChangeset` and `Identifiable`
+///
+/// `foo.save_changes(&conn)` is equivalent to
+/// `update(foo::table().find(foo.id())).set(&foo).get_result(&conn)`
+///
+/// # Example
+///
+/// ```rust
+/// # #[macro_use] extern crate diesel;
+/// # #[macro_use] extern crate diesel_codegen;
+/// # include!("../doctest_setup.rs");
+/// #
+/// # table! {
+/// #     users {
+/// #         id -> Integer,
+/// #         name -> VarChar,
+/// #     }
+/// # }
+/// #
+/// # #[derive(Debug, PartialEq, Identifiable, Queryable)]
+/// # struct Animal {
+/// #    id: i32,
+/// #    species: String,
+/// #    legs: i32,
+/// #    name: Option<String>,
+/// # }
+/// #
+/// # #[derive(AsChangeset, Identifiable)]
+/// # #[table_name = "animals"]
+/// # pub struct UpdateAnimal<'a> {
+/// #   id: i32,
+/// #   species: Option<&'a str>,
+/// #   legs: Option<i32>,
+/// #   name: Option<&'a str>,
+/// # }
+/// #
+/// # fn main() {
+/// #     use animals::dsl::*;
+/// #     let connection = establish_connection();
+/// let spider = animals.filter(species.eq("spider"))
+///     .first::<Animal>(&connection)
+///     .expect("Too scary to load");
+///
+/// let not_a_spider = UpdateAnimal {
```

Comment:
This example misses the point of this API. While it may sometimes be used with a separate update struct, most usage of this would have `#[derive(AsChangeset)]` on the main struct. e.g: 

```rust
let mut spider = animals.filter(species.eq("spider")
    .first::<Animal>(&connection)
    .expect("Too scary to load");

spider.spieces = String::from("solifuge");
spider.legs = 10;

spider = spider.save_changes(&connection).expect("Could not save changes");
```

Reply:
Awesome! I misunderstood the current docs. I'll make note of this as well.